src/main/java/soonwee/duke/Duke.java

```diff
@@ -0,0 +1,160 @@
+package soonwee.duke;
+
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represents a Duke instance. A Duke instance will prompt user which bot they
+ * want to use (1. Echo Bot, 2. Tasks List Bot).
+ */
+public class Duke {
+
+    public static final int TODO = 1;
+    public static final int DEADLINE = 2;
+    public static final int EVENT = 3;
+    public static final int FIND = 4;
+    public static Ui display = new Ui();
+
+    /**
+     * Starts the program.
+     */
+    public static void main(String[] args) throws IOException {
+        System.out.println(display.printGreeting());
+        System.out.println(display.printPrompt());
+        selectAction();
+        System.out.println(display.printBye());
+    }
+
+    /**
+     * Prompts the user which bot they want to choose.
+     *
+     * @throws IOException if an error occurs.
+     */
+    public static void selectAction() throws IOException {
+        BufferedReader input = new
+                BufferedReader(new InputStreamReader(System.in));
+        String cmd = input.readLine();
+        switch (cmd) {
+        case "1":
+            performFirstTask();
+            break;
+        case "2":
+            TaskList temp = performSecondTask();
+            break;
+        default:
+            break;
+        }
+        input.close();
+    }
+
+    /**
+     * Performs the first macro task which will echo whatever the user types
+     * in and ends when the user types 'bye'.
+     *
+     * @throws IOException if an error occur.
+     */
+    public static void performFirstTask() throws IOException {
+        String cmd = new String();
+        BufferedReader input = new
+                BufferedReader(new InputStreamReader(System.in));
+        while (true) {
+            cmd = input.readLine();
+            System.out.println(display.printLine());
+            if (cmd.equals("bye")) {
+                break;
+            }
+            System.out.println(cmd);
+            System.out.println(display.printLine());
+        }
+    }
+
+    /**
+     * Performs the second macro task which can add, delete or mark a task as
+     * done. Macro will end when user types in 'bye'.
+     *
+     * @return updated Task List.
+     * @throws IOException if file or directory is not found.
+     */
+    public static TaskList performSecondTask() throws IOException {
+        String cmd = new String();
+        Storage storage = new Storage("C:\\Users\\songs\\Desktop\\CS2103\\data\\tasks.txt");
```

Comment:
Just curious whether using absolute path will influence future distribution of the program.